hooks/tk-multi-publish2/tk-maya/unreal/unreal_setup_turntable.py

```
@@ -0,0 +1,165 @@
+    scale_x = 250 / min(bounds.x, bounds.y)
+    scale_y = 300 / max(bounds.x, bounds.y)
+    scale_z = 200 / bounds.z
+    scale = min(scale_x, scale_y, scale_z)
+    actor.set_actor_scale3d(unreal.Vector(scale, scale, scale))
+    
+    # Offset the actor location so that it rotates around its center
+    origin = origin * scale
+    actor.set_actor_location(unreal.Vector(-origin.x, -origin.y, -origin.z), False, True)
+    
+    # Attach the newly spawned actor to the turntable
+    actor.attach_to_actor(turntable_actor, "", unreal.AttachmentRule.KEEP_WORLD, unreal.AttachmentRule.KEEP_WORLD, unreal.AttachmentRule.KEEP_WORLD, False)
+
+    unreal.log("Saving current level...")
+    unreal.EditorLevelLibrary.save_current_level()
+    unreal.log("Saving map %s" % world.get_path_name())
+    # This seems to fail with:
+    # [2021.09.15-15.17.52:220][  1]Message dialog closed, result: Ok, title: Message, text: Failed to save the map. The filename '../../../../../../../var/folders/rt/vlgl5dzj75q2qg4t9fp9gfx80000gp/T/tmpO8A5Lw/turntable/Content/turntable/level/turntable.turntable.umap' is not within the game or engine content folders found in '/Users/Shared/Epic Games/UE_4.26/'.
+    unreal.EditorLoadingAndSavingUtils.save_map(world, world.get_path_name())
+    unreal.log("Turntable setup done.")
+    return world.get_path_name()
+
+if __name__ == "__main__":
+    # Script arguments must be, in order:
+    # Path to FBX to import
+    # Unreal content browser path where to store the imported asset
+    # Unreal content browser path to the turntable map to duplicate and where to spawn the asset
+    # Retrieve arguments from the environment
+    fbx_file_path = os.environ["UNREAL_SG_FBX_OUTPUT_PATH"]
+    assets_path = os.environ["UNREAL_SG_ASSETS_PATH"]
+    turntable_map_path = os.environ["UNREAL_SG_MAP_PATH"]
+
+    # Additional settings to render with the Movie Render Queue
+    movie_path = os.environ.get("UNREAL_SG_MOVIE_OUTPUT_PATH")
```

Just a heads up: If the environment variable doesn't exists the returned value will be None and maybe you need to raise an exception in that case?

---

no because these ones are optional.